File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldRefReport.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdGrid;
import com.neome.api.meta.base.Types.MetaIdReport;

import org.jetbrains.annotations.Nullable;

import java.util.Map;

@SuppressWarnings("unused")
public class DefnFieldRefReport extends DefnField
{
  @Nullable
  public Map<MetaIdField, MetaIdField> copyFieldMap;

  @Nullable
  public MetaIdField[] editableFieldIdSet;

  @Nullable
  public Boolean forceOpenOnFormCreate;

  @Nullable
  public Boolean forceOpenOnGridRowCreate;

  @Nullable
  public MetaIdGrid gridId;

  @Nullable
  public MetaIdField[] keyFieldIdSet;

  @Nullable
  public DefnLayoutGrid layoutGrid;

  @Nullable
  public DefnDtoLayoutOverlaySpreadsheet overlayLayoutGrid;

  public MetaIdReport reportId;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldRefReport.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.dto.DefnDtoLayoutOverlaySpreadsheet
import com.neome.api.meta.base.dto.DefnField
import com.neome.api.meta.base.dto.DefnLayoutGrid
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdGrid
import com.neome.api.meta.base.Types.MetaIdReport

interface DefnFieldRefReport : DefnField
{
  val copyFieldMap: Map<MetaIdField, MetaIdField>?
  val editableFieldIdSet: List<MetaIdField>?
  val forceOpenOnFormCreate: Boolean?
  val forceOpenOnGridRowCreate: Boolean?
  val gridId: MetaIdGrid?
  val keyFieldIdSet: List<MetaIdField>?
  val layoutGrid: DefnLayoutGrid?
  val mobileLayoutGrid: DefnLayoutGrid?
  val mobileOverlayLayoutGrid: DefnDtoLayoutOverlaySpreadsheet?
  val overlayLayoutGrid: DefnDtoLayoutOverlaySpreadsheet?
  val reportId: MetaIdReport
}